oslo/torch/nn/parallel/data_parallel/zero/utils.py

```diff
@@ -0,0 +1,12 @@
+import torch
+
+
+def get_current_device() -> torch.device:
+    """
+    Returns currently selected device (gpu/cpu).
+    If cuda available, return gpu, otherwise return cpu.
+    """
+    if torch.cuda.is_available():
+        return torch.device(f"cuda:{torch.cuda.current_device()}")
+    else:
+        return torch.device("cpu")
```

Comment:
How about just using `torch.cuda.current_device()`?
There are already many lines that do not check the CUDA availability in oslo.

Or it will be better to integrate theses similar functions. https://github.com/EleutherAI/oslo/blob/2f6d2a9287e6832d7f950c570fc12acf07ff4ccc/oslo/torch/nn/parallel/expert_parallel/utils.py#L8-L12

Reply:
Thanks to you, I've realized that there are redundant functions for obtaining information about the currently used device. I appreciate your help. However, I believe it is not appropriate to consolidate them in the current pull request. Instead, I will create a new pull request to merge the device query logic scattered across various parallelization methodologies. Does that sound good to you?